jumpscale/sals/process/__init__.py

```diff
@@ -84,595 +141,833 @@ def execute(
+                yield proc
+    except (psutil.AccessDenied, psutil.NoSuchProcess):
+        pass
 
 
-def check_running(process, min=1):
-    """Checks if a process is running
+def check_running(process_name, min=1):
+    """Check if there are a specific number of running processes that match the given name.
 
-    Arguments:
-        process {str} -- process name to be checked
+    Function will check string against Process.name(), Process.exe() and Process.cmdline().
 
-    Keyword Arguments:
-        min (int) -- min number of instances required to be running (default: {1})
+    Args:
+        process_name (str): the target process name
+        min (int, optional): min number of instances required to be running. Defaults to 1.
 
     Returns:
-        [bool] -- true if process is running
+        bool: true if process is running, otherwise False
     """
-    if j.data.platform.is_linux():
-        pids = get_pids(process)
-        if len(pids) >= min:
-            return True
-        return False
+    pids = get_pids(process_name, limit=min)
+    return len(pids) == min
 
 
 def check_process_for_pid(pid, process_name):
     """Check whether a given pid actually does belong to a given process name.
 
-    Arguments:
-        pid (int) -- process pid
-        process {str} -- process name
+    Args:
+        pid (int): Process ID
+        process (str): String to match againset candidate processes name using equality operator
 
     Returns:
-        [bool] -- True if process_name matched process name of the pid
+        bool: True if process_name matched process name of the pid, False otherwise.
     """
-    pid = int(pid)
-    proc = psutil.Process(pid)
-    return proc.name() == process_name
+    try:
+        proc = psutil.Process(pid)
+        return proc.name() == process_name
+    except (psutil.AccessDenied, psutil.NoSuchProcess):
+        return False
 
 
-def set_env_var(varnames, varvalues):
-    """Set the value of the environment variables C{varnames}. Existing variable are overwritten
+def set_env_var(var_names, var_values):
+    """Set the value of the environment variables {varnames}. Existing variable are overwritten
 
-    Arguments:
-        varnames {list(str)} --  A list of the names of all the environment variables to set
-        varvalues {list(str)} -- A list of all values for the environment variables
+    Args:
+        var_names list[str]: A list of the names of all the environment variables to set
+        varvalues list[str]: A list of all values for the environment variables
 
+    Raises:
+        j.exceptions.RuntimeError: if error happened during setting the environment variables
     """
     try:
-        for i in range(len(varnames)):
-            os.environ[varnames[i]] = str(varvalues[i]).strip()
+        for i in range(len(var_names)):
+            os.environ[var_names[i]] = str(var_values[i]).strip()
     except Exception as e:
```

Comment:
Can we handle expected exceptions types here instead of masking all types to a `RuntimeError`?

Reply:
after checking `os.environ` documentation, I can't think of any exception that code could throw.
so, I removed it.